compiler/daml-lf-reader/src/DA/Daml/LF/Reader.hs

```diff
@@ -0,0 +1,43 @@
+-- Copyright (c) 2019 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+-- SPDX-License-Identifier: Apache-2.0
+
+module DA.Daml.LF.Reader
+    ( Manifest(..)
+    , ManifestData(..)
+    , manifestFromDar
+    ) where
+
+import Codec.Archive.Zip
+import qualified Data.ByteString.Lazy as BSL
+import qualified Data.ByteString.Lazy.UTF8 as UTF8
+import qualified Data.HashMap.Strict as Map
+import Data.List.Extra
+import System.FilePath
+
+data Manifest = Manifest { mainDalf :: FilePath , dalfs :: [FilePath] } deriving (Show)
+data ManifestData = ManifestData { mainDalfContent :: BSL.ByteString , dalfsCotent :: [BSL.ByteString] } deriving (Show)
+
+lineToKeyValue :: String -> (String, String)
+lineToKeyValue line = case splitOn ":" line of
+    [l, r] -> (trim l , trim r)
+    _ -> ("malformed", "malformed")
```

Comment:
Not a fan of sneaking `malformed` into places - you're one day going to get an error that just talks about malformed... Perhaps just silently drop them with a Nothing?

Reply:
I’ve changed it to a call to `error` for now, I’m not a fan of silently dropping on parse failures.